commands/template_pull_test.go

```diff
@@ -4,141 +4,123 @@ package commands
-		t.Fatalf("The directory %s was not created", "template")
-	}
-}
+	t.Run("WithOverwriting", func(t *testing.T) {
+		defer tearDownFetchTemplates(t)
 
-func Test_templatePull_no_arg(t *testing.T) {
-	defer tearDown_fetch_templates(t)
-	var buf bytes.Buffer
+		faasCmd.SetArgs([]string{"template", "pull", localTemplateRepository})
+		faasCmd.Execute()
 
-	faasCmd.SetArgs([]string{"template", "pull"})
-	faasCmd.SetOutput(&buf)
-	faasCmd.Execute()
+		var buf bytes.Buffer
+		log.SetOutput(&buf)
 
-	if strings.Contains(buf.String(), "Error: A repository URL must be specified") {
-		t.Fatal("Output does not contain the required string")
-	}
-}
+		r := regexp.MustCompile(`(?m:Cannot overwrite the following \d+ template\(s\):)`)
 
-func Test_templatePull_error_not_valid_url(t *testing.T) {
-	var buf bytes.Buffer
+		faasCmd.SetArgs([]string{"template", "pull", localTemplateRepository})
+		faasCmd.Execute()
 
-	faasCmd.SetArgs([]string{"template", "pull", "git@github.com:openfaas/faas-cli.git"})
-	faasCmd.SetOutput(&buf)
-	err := faasCmd.Execute()
+		if !r.MatchString(buf.String()) {
+			t.Fatal(buf.String())
+		}
 
-	if !strings.Contains(err.Error(), "The repository URL must be in the format https://github.com/<owner>/<repository>") {
-		t.Fatal("Output does not contain the required string", err.Error())
-	}
-}
+		buf.Reset()
 
-// httpTestServer returns a testing http server
-func httpTestServer(t *testing.T) *httptest.Server {
-	const sampleMasterZipPath string = "testdata/master_test.zip"
-	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		faasCmd.SetArgs([]string{"template", "pull", localTemplateRepository, "--overwrite"})
+		faasCmd.Execute()
 
-		if _, err := os.Stat(sampleMasterZipPath); os.IsNotExist(err) {
-			t.Error(err)
+		str := buf.String()
+		if r.MatchString(str) {
+			t.Fatal()
 		}
 
-		fileData, err := ioutil.ReadFile(sampleMasterZipPath)
-		if err != nil {
-			t.Error(err)
+		// Verify created directories
+		if _, err := os.Stat("template"); err != nil {
+			t.Fatalf("The directory %s was not created", "template")
 		}
+	})
 
-		w.Write(fileData)
-	}))
+	t.Run("InvalidUrlError", func(t *testing.T) {
+		var buf bytes.Buffer
 
-	return ts
-}
+		faasCmd.SetArgs([]string{"template", "pull", "user@host.xz:openfaas/faas-cli.git"})
+		faasCmd.SetOutput(&buf)
+		err := faasCmd.Execute()
 
-func Test_repositoryUrlRegExp(t *testing.T) {
-	var url string
-	r := regexp.MustCompile(repositoryRegexpGithub)
+		if !strings.Contains(err.Error(), "The repository URL must be a valid git repo uri") {
+			t.Fatal("Output does not contain the required string", err.Error())
+		}
+	})
+}
 
-	url = "http://github.com/owner/repo"
-	if r.MatchString(url) {
-		t.Errorf("Url %s must start with https", url)
+func Test_repositoryUrlRemoteRegExp(t *testing.T) {
+
+	r := regexp.MustCompile(gitRemoteRepoRegex)
+	validURLs := []string{
+		"git://github.com/openfaas/faas.git#ff78lf9h",
```

Comment:
Please can I ask you to look at how we use test-case tables to name scenarios?

It may help reduce the scenarios we maintain if we name them too.

i.e. 

```
{ "scenario": "branch", "url" : "git://github.com/openfaas/faas.git#quick_fix"},
{ "scenario": "ssh without port", "url" : "ssh://user@host.xz/path/to/repo.git/"},
```